### PR TITLE
Add ability for custom package names

### PR DESCRIPTION
**Pull request by tampakrap**
Allow overriding the default distro package names by specifying
$puppet::agent::package and $puppet::server::package

### CODE REVIEW

**Comment by igalic** [comment]
sitting in the puppet-server talk by @KevinCorcoran's talk and… now i want this.

**Comment by danieldreier** [comment]
+1

File: manifests/package/gentoo.pp [hunk 1]
```diff
@@ -1,16 +1,15 @@
 class puppet::package::gentoo {
 
   include puppet::agent
-  include puppet::params
 
   if $puppet::server::master {
     include puppet::server
     $keywords = $puppet::server::gentoo_keywords
-    $package  = $puppet::params::master_package
+    $package  = $puppet::server::package
```
Comment (xaque208) [inline manifests/package/gentoo.pp]: This variable isn't guaranteed to be in scope.

**Comment by xaque208** [comment]
How about https://github.com/puppetlabs-operations/puppet-puppet/pull/177 instead?

**Comment by tampakrap** [comment]
Yes I like your approach very much! Let's continue there

**Pull request closed without merging**